### PR TITLE
trace patch operations

### PR DESCRIPTION
**Pull request by CaoShuFeng**
Just like `update`, `create`, `get` and `delete` operations.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go [hunk 1]
```diff
@@ -168,6 +177,9 @@ func patchResource(
 	// and is given the currently persisted object as input.
 	applyPatch := func(_ request.Context, _, currentObject runtime.Object) (runtime.Object, error) {
 		// Make sure we actually have a persisted currentObject
+		if trace != nil {
```
Comment (enj) [inline staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go]: It seems more reasonable to require a non nil trace.

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest_test.go [hunk 2]
```diff
@@ -347,7 +347,7 @@ func (tc *patchTestCase) Run(t *testing.T) {
 			name,
 			patchType,
 			patch,
-			namer, creater, defaulter, convertor, kind, resource, codec)
+			namer, creater, defaulter, convertor, kind, resource, codec, nil)
```
Comment (enj) [inline staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest_test.go]: So I would create one here instead of passing nil.

**Review by enj — CHANGES_REQUESTED** [review]

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go (hunk 1 above)
Comment (CaoShuFeng) [reply staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go]: Done.

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest_test.go (hunk 2 above)
Comment (CaoShuFeng) [reply staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest_test.go]: Done.

**Comment by enj** [comment]
/lgtm

Code looks fine.  @smarterclayton can probably judge if we need this or not (or if it was left out on purpose).

**Comment by CaoShuFeng** [comment]
/assign @sttts 
assign @sttts for some suggestions.

**Comment by sttts** [comment]
Also looks fine for me. But I wonder who owns the tracing code. @smarterclayton? @wojtek-t?

**Pull request merged**